hypothesis-python/tests/ghostwriter/test_ghostwriter_cli.py

```diff
@@ -116,6 +122,139 @@ def test_can_import_from_scripts_in_working_dir(tmpdir):
     assert "Error: " not in result.stderr
 
 
+CLASS_CODE_TO_TEST = """
+from typing import Sequence, List
+
+def my_func(seq: Sequence[int]) -> List[int]:
+    return sorted(seq)
+
+class MyClass:
+
+    @staticmethod
+    def my_staticmethod(seq: Sequence[int]) -> List[int]:
+        return sorted(seq)
+
+    @classmethod
+    def my_classmethod(cls, seq: Sequence[int]) -> List[int]:
+        return sorted(seq)
+"""
+
+
+@pytest.mark.parametrize("func", ["my_staticmethod", "my_classmethod"])
+def test_can_import_from_class(tmpdir, func):
+    (tmpdir / "mycode.py").write(CLASS_CODE_TO_TEST)
+    result = subprocess.run(
+        f"hypothesis write mycode.MyClass.{func}",
+        capture_output=True,
+        shell=True,
+        text=True,
+        cwd=tmpdir,
+    )
+    assert result.returncode == 0
+    assert "Error: " not in result.stderr
+
+
+@pytest.mark.parametrize(
+    "classname,funcname,err_msg",
+    [
+        (
+            "XX",
+            "XX",
+            "Found the 'mycode' module, but it doesn't have a 'XX' class.",
+        ),
+        (
+            "MyClass",
+            "XX",
+            "Found the 'mycode' module and 'MyClass' class, but it doesn't have a 'XX' attribute.",
+        ),
+        (
+            "my_func",
+            "XX",
+            "Found the 'mycode' module and 'my_func' attribute, but it doesn't have a 'XX' attribute.",
+        ),
```

Comment:
You could compress this parametrize down to just `("my_func", " and 'my_func' attribute", "attribute")` - with a bit more string formatting.  That would make it much more compact, which I think would be easier to read.

Reply:
Do you mean the error message? Can't quite get what you mean here 🙃